[backport to 2.7] bpo-28929: Link the documentation to its source file on GitHub

Mariatta opened the pull request:
Change the documentation's `Show Source` link on the left menu
to GitHub source file.

brettcannon commented:
Thanks for the PR, but https://github.com/python/cpython is a cpython mirror repository; pull requests are not accepted on this repo. Please submit bugs and patches at https://bugs.python.org.

Mariatta committed:
bpo-28929: Link the documentation to its source file on GitHub

Change the documentation's `Show Source` link on the left menu
to GitHub source file.
